sw/device/sca/sha3_serial.c

```diff
@@ -471,6 +500,52 @@ static void sha3_serial_single_absorb(const uint8_t *msg, size_t msg_len) {
   kmac_reset();
 }
 
+static void sha3_serial_fixed_key_set(const uint8_t *key, size_t key_len) {
+  SS_CHECK(key_len == kKeyLength);
+  memcpy(key_fixed, key, key_len);
+}
+
+static void sha3_serial_batch(const uint8_t *data, size_t data_len) {
+  uint32_t num_encryptions = 0;
+  uint32_t out[kDigestLength];
+  uint32_t batch_digest[kDigestLength];
+  SS_CHECK(data_len == sizeof(num_encryptions));
+  num_encryptions = read_32(data);
+
+  // This bit may not be needed. Test afterwards
+  for (uint32_t j = 0; j < kDigestLength; ++j) {
+    batch_digest[j] = 0;
+  }
```

Comment:
Is it needed or not?

Reply:
I've tested it and it is indeed needed. Comment removed.